common/lib/dependabot/metadata_finders/base.rb

```diff
@@ -11,99 +11,51 @@ class Base
 
       PACKAGE_MANAGERS_WITH_RELIABLE_DIRECTORIES = %w(npm_and_yarn pub).freeze
 
-      attr_reader :dependency, :credentials
+      attr_reader :dependency, :credentials, :changelog_url, :changelog_text, :upgrade_guide_url, :upgrade_guide_text,
+                  :releases_url, :releases_text, :commits_url, :commits
 
       def initialize(dependency:, credentials:)
         @dependency = dependency
         @credentials = credentials
-      end
-
-      def source_url
```

Comment:
We chatted about this internally, but wanted to mention here publicly as well... moving everything into the initialization just seems a bit off... suddenly instantiating an instance requires network I/O and if I want only a single method or two, I now have to get everything... I wasn't the only person with some hesitations about this path forward, so the problem/debugging is awesome, but we may end up pursuing a different/more flexible solution.

Reply:
You're absolutely right!